code/obj/artifacts/artifactanalysis.dm

```diff
@@ -183,3 +183,14 @@
 			var/obj/O = src.attached
 			O.remove_suffixes("\[[src.artifactType]\]")
 			O.UpdateName()
+
+	mouse_drop(atom/over_object, src_location, over_location, over_control, params)
+		if(!istype(usr, /mob/living) || !isturf(src.loc) || \
+				BOUNDS_DIST(get_turf(over_object), get_turf(src)) > 0 || \
+				BOUNDS_DIST(usr, get_turf(over_object)) > 0 ||  \
+				BOUNDS_DIST(usr, src) > 0 || \
+				over_object == usr || !istype(over_object, /atom/movable))
+			return ..()
+		var/atom/movable/target = over_object
+		usr.visible_message("<span class='notice'>[usr] sticks a [src.name] on [target].</span>")
+		src.stick_to(target, src.pixel_x, src.pixel_y, usr)
```

Comment:
```suggestion
		src.stick_to(target, text2num(params["icon-x"]) - 16, text2num(params["icon-y"]) - 16, usr)
```
`params` holds info about where you dragged to; this should make it so the form gets stuck to exactly where you dragged it to, instead of just keeping the same pixel position as it had pre-drag.
(stolen from `/obj/proc/place_on`)